### ISSUE TITLE
Use Github Actions

### ISSUE THREAD

**Issue by lucc** [pazz/alot#1616]
**Is your feature request related to a problem? Please describe.**
The travis CI is non functional for quite some time and it would be good to have CI again. 

**Describe the solution you'd like**
I have set up github actions for some other repos already and want to do it.

**Describe alternatives you've considered**
none, I did think about travis some years back when they changed their terms of service but decided to use GHA with nvimpager and khard.

**Additional context**

https://github.com/pazz/alot/pull/1615#issuecomment-1508667410

@pazz you can drop some ideas here about what you want and then we can discuss the tasks that should happen in CI.

**Comment by pazz** [comment]
Many thanks @lucc !
I am on the notmuch list but only (rarely) look at python related posts. Somehow, the python bindings seem to change "maintainers" frequently but I don't think they have set up a hook to publish current versions on pypi.
Perhaps we should send a PR there or suggest that they automate this on notmuch version release.

One thing that could go into the GHAs here, what we did on travis IIRC, is building the (sphinx) docs and see if that fails...

**Comment by pazz** [comment]
See [here](https://github.com/pazz/alot/blob/de39276b0915a794b91018d0807a101b38eaf61d/.travis.yml) for the most recent travis config I could find..

**Comment by lucc** [comment]
@pazz yes please send a request to the list that they at least publish the current version on PyPI. Having them published automatically would be the optimal solution for us in the future.

**Comment by teto** [comment]
would ne nice restoring the pipelines, for something as critical as a mailreader :) let me know if I can review. One solution could be to just run a nix build once https://github.com/pazz/alot/pull/1624 is finished/merged. A new GHA action seems to autocache the builds
